docs/website/docs/dlt-ecosystem/verified-sources/openapi-generator.md

```diff
@@ -0,0 +1,198 @@
+---
+title: OpenAPI source generator
+description: dlt verified source for Personio API
+keywords: [personio api, personio verified source, personio]
+---
+import Header from './_source-info-header.md';
+
+# OpenAPI source generator
+
+<Header/>
+
+Our OpenAPI source generator - `dlt-init-openapi` - generates [`dlt`](https://dlthub.com/docs) data pipelines from OpenAPI 3.x specs using the [dlt rest_api verified source](https://dlthub.com/docs/dlt-ecosystem/verified-sources/rest_api) to extract data from any REST API. If you are not familiar with the `rest_api` source, please read [dlt rest_api](https://dlthub.com/docs/dlt-ecosystem/verified-sources/rest_api) to learn how our `rest_api` source works.
+
+:::tip
+We also have a cool [Google Colab example](https://colab.research.google.com/drive/1MRZvguOTZj1MlkEGzjiso8lQ_wr1MJRI?usp=sharing#scrollTo=LHGxzf1Ev_yr) that demonstrates this generator. 😎
+:::
+
+## Features
+`dlt-init-openapi` generates code from an OpenAPI spec that you can use to extract data from a `rest_api` into any [`destination`](https://dlthub.com/docs/dlt-ecosystem/destinations/) (e.g., Postgres, BigQuery, Redshift...) that `dlt` supports. dlt-init-openapi additionally executes a set of heuristics to discover information not explicitly defined in OpenAPI specs.
+
+Features include:
+
+* **[Pagination](https://dlthub.com/docs/dlt-ecosystem/verified-sources/rest_api#pagination) discovery** for each endpoint.
+* **Primary key discovery** for each entity.
+* **Endpoint relationship mapping** into `dlt` [`transformers`](https://dlthub.com/docs/general-usage/resource#process-resources-with-dlttransformer) (e.g., /users/ -> /user/{id}).
+* **Payload JSON path [data selector](https://dlthub.com/docs/dlt-ecosystem/verified-sources/rest_api#data-selection) discovery** for results nested in the returned JSON.
+* **[Authentication](https://dlthub.com/docs/dlt-ecosystem/verified-sources/rest_api#authentication)** discovery for an API.
+
+## A quick example
+
+You will need Python 3.9 or higher installed, as well as pip. You can run `pip install dlt-init-openapi` to install the current version.
+
+We will create a simple example pipeline from a [PokeAPI spec](https://pokeapi.co/) in our repo. You can point to any other OpenAPI Spec instead if you prefer.
+
+```sh
+# 1.a. Run the generator with a URL:
```

Comment:
I'd make this a regular text (not a shell code) and make each actual commands a code snippet (removing the prompt "$" sign so it's easy to copy & paste)

Reply:
imo this will improve the readability + make copy and paste more convenient.

Reply:
yeah, on docusaurus its better this way! changed